templates/test/ci/prow-flatcar/cni-resource-set.yaml

```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: addons.cluster.x-k8s.io/v1beta1
+kind: ClusterResourceSet
+metadata:
+  name: ${CLUSTER_NAME}-calico
```

Comment:
we recently removed all ClusterResourceSets to install calico (see #2495). So here's what you need to do:

1. Rebase on top of latest main
2. Get rid of all the Calico stuff in the template/kustomization
3. Add 
``` 
				ControlPlaneWaiters: clusterctl.ControlPlaneWaiters{
					WaitForControlPlaneInitialized: EnsureControlPlaneInitialized,
				},
````

in azure_test.go in the `ApplyClusterTemplateAndWaitInput` (see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2890/files#diff-93ce65d81f5c1510135f6035d43f678bf9b5962b8e4ee45105e3345a1852efc6L321 for an example)

Reply:
Thanks again for the hint. I was suspecting something has changed but I did not investigate it further. Changes are done. Tests seems to be passing locally. Please have a look :)